docs: adds pros n cons for access methods (#636)
Azure/secrets-store-csi-driver-provider-azure: website/content/en/configurations/identity-access-modes/_index.md

weight: 1
description: >
  The Azure Key Vault Provider offers four modes for accessing a Key Vault instance
---

> ## Best Practices:

Reviewer: ```suggestion
## Best Practices

```

Following order of access modes is recommended for Secret Store CSI driver AKV provider:

| Access Option 	| Comment 	|
|---	|---	|

> | Pod Identity 	| This is the most secure way to get access to Azure resources (AKV in this case) as it's restricted for a given Pod. 	|

Reviewer: ```suggestion
| Pod Identity 	| This is the most secure way to get access to Azure resources (AKV in this case) as it uses the managed identity bound to the Pod. 	|

```

> | Managed Identities (System-assigned and User-assigned) 	| AKS using managed identities manages credentials required to connect to Azure Resources. 	|

Reviewer: ```suggestion
| Managed Identities (System-assigned and User-assigned) 	| Managed identities eliminate the need for developers to manage credentials. Managed identities provide an identity for applications to use when connecting to Azure Keyvault. 	|

nit: managed identities are not specific to aks

```

> | Service Principal 	| This is the last option to consider while connecting to AKV as access credentials need to be created as k8s Secret and stored in plain text in etcd.<br>Also, this is the only option to connect to Azure resources from non Azure environment/cluster. 	|

Reviewer: ```suggestion
| Service Principal 	| This is the last option to consider while connecting to AKV as access credentials need to be created as Kubernetes Secret and stored in plain text in etcd.<br>Also, this is the only option to connect to Azure resources from non Azure environment/cluster. 	|
```
